### ISSUE TITLE
Can Seashore be a macOS Photos extension?

### ISSUE THREAD

**Issue by davidhercher** [robaho/seashore#214]
I'd like to use Seashore to edit photos in macOS Photos by using Seashore as an extension. I have the current macOS (15.0) and Photos (1.0, 701.0.220). In Photos, after selecting a photo and then choosing Edit | three dots in the tool bar, I see Markup above the line and App Store and Manage below it. Choosing App Store identifies only seven Mac App Store apps as candidates to use as a Photos Extension, and Seashore is not one of the seven, even though it's available in the Mac App store. When I choose Manage, I see a list of extensions, including extensions to Spotlight, which has a list including Seashore. But I don't see how actually to see Seashore within Photos. 

Can Seashore be used in Photos as an extension? Or is it usable only outside Photos? Thanks.

**Comment by robaho** [comment]
Thanks for reporting this. This used to work. I’ll get it fixed this week.

**Comment by davidhercher** [comment]
That’s great. Would you expect it to be on the list of available extensions when choosing App Store? I’m surprised that only seven appear there. Thanks. David Hercher From: robert engels ***@***.***>Date: Friday, September 20, 2024 at 3:36 PMTo: robaho/seashore ***@***.***>Cc: davidhercher ***@***.***>, Author ***@***.***>Subject: Re: [robaho/seashore] Can Seashore be a macOS Photos extension? (Issue #214)Thanks for reporting this. This used to work. I’ll get it fixed this week.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by robaho** [comment]
No. It will be available with “edit with”. I may be able to support the App Store but I’d have to look into it.

**Comment by robaho** [comment]
Also your email replies are corrupted. I needed to edit it. Please update the issue using a browser.

**Comment by robaho** [comment]
I did notice one issue, if you use Edit, then close and save at the close, it doesn't work. You need to manually Save in the editor first. I will look into this.

**Comment by davidhercher** [comment]
Thank you for your time on this project. Two remaining questions: (1) After saving changes made in Seashore, is it possible to revert to the original image, as it is after editing using Photos itself; and (2) Is it correct that Seashore, although available in the Mac App Store, can't be added to Photos' internal extension list via selecting photo | Edit | three dots | App Store. When I do the latter, I see seven apps on the Mac App Store, not including Seashore. If one of the seven is chosen, it appears in the drop-down extension list under the three dotes. Thanks again.

**Comment by robaho** [comment]
1. It doesn't appear so. I believe this used to be the case but I think Apple has removed this feature. The "Revert To Original" is grayed out. I am going to look a bit more into this to see if there are options to turn it on.
2. As I mentioned above, a Photos Extension requires OSX 10.11 to implement which is outside the minimum Seashore version. I'm not saying it would be impossible to implement, but it will be a bit of effort, but at this time Seashore is not a Photos Extension.

**Comment by robaho** [comment]
I have done more work on this. The photo's extension isn't really applicable to integration with Seashore. The "edit with" is what you want. I have tested, and the only issue is that you need to use file save, "closing at save" will not persist the changes.

**Comment by robaho** [comment]
Just a note, the replacement Seahorse has fully working 'Revert to Last Saved' and version browsing. I am really not sure why this stopped working in Seashore.
